sdks/go/pkg/beam/core/runtime/exec/coder.go

```diff
@@ -1208,3 +1213,91 @@ func DecodeWindowedValueHeader(dec WindowDecoder, r io.Reader) ([]typex.Window,
 
 	return ws, t, pn, nil
 }
+
+// EncodeTimer encodes a typex.TimerMap into a byte stream.
+func EncodeTimer(elm ElementEncoder, tm typex.TimerMap, w io.Writer) error {
+	var b bytes.Buffer
+
+	elm.Encode(&FullValue{Elm: tm.Key}, &b)
+
+	if err := coder.EncodeStringUTF8(tm.Tag, &b); err != nil {
+		return errors.WithContext(err, "error encoding tag")
+	}
+
+	if _, err := ioutilx.WriteUnsafe(&b, tm.Windows); err != nil {
+		return err
+	}
+
+	if err := coder.EncodeBool(tm.Clear, &b); err != nil {
+		return errors.WithContext(err, "error encoding key")
+	}
+
+	if !tm.Clear {
+		if err := coder.EncodeVarInt(tm.FireTimestamp, &b); err != nil {
```

Comment:
VarInt isn't the correct encoding for the FireTimestamp or the HoldTimestamp

See
https://github.com/apache/beam/blob/master/model/pipeline/src/main/proto/org/apache/beam/model/pipeline/v1/beam_runner_api.proto#L873

It should be coder.EncodeEventTime instead: https://github.com/apache/beam/blob/master/sdks/go/pkg/beam/core/graph/coder/time.go#L30

Reply:
Thanks for catching this. The big endian integer got me the other way.